ports/rp2/tusb_port.c

```diff
@@ -102,9 +103,22 @@ const uint16_t *tud_descriptor_string_cb(uint8_t index, uint16_t langid) {
         if (index >= sizeof(usbd_desc_str) / sizeof(usbd_desc_str[0])) {
             return NULL;
         }
-        const char *str = usbd_desc_str[index];
-        for (len = 0; len < DESC_STR_MAX - 1 && str[len]; ++len) {
-            desc_str[1 + len] = str[len];
+        // check, if serial is required
+        if (index == USBD_STR_SERIAL) {
+            pico_unique_board_id_t id;
+            pico_get_unique_board_id(&id);
+            // byte by byte conversion
+            char temp_str[3];
+            for (len = 0; len < 16; len += 2) {
+                sprintf(temp_str, "%02x", id.id[len >> 1]);
+                desc_str[1 + len] = temp_str[0];
+                desc_str[1 + len + 1] = temp_str[1];
```

Comment:
It's possibly a bit heavy to use `sprintf` to get these two characters (if you do use this function, use `snprintf`).  Could be simpler to do:
```c
const char *hexdig = "0123456789abcdef";
desc_str[1 + len] = hexdig[id.id[len >> 1] >> 4];
desc_str[1 + len + 1] = hexdig[id.id[len >> 1] & 0x0f];
```
or see `ports/stm32/mphalport.c:mp_hal_get_mac_ascii` for another way.

Reply:
I am alittle bit unhappy about sprintf usage too, but assumed, that since printf is already used, the code is linked anyhow, and the size overhead could be smaller than the 16 bytes used for the hexdig string. But I can change that easily.

Reply:
changed